demo-contract-tests/tests/test_fixture/wasm_helper.rs

```diff
@@ -0,0 +1,78 @@
+use std::env;
+use std::fs;
+use std::path::Path;
+use std::path::PathBuf;
+
+pub fn get_wasm_directory() -> PathBuf {
+    // Environment variable or default path.
+    let base_path = if let Ok(custom_path) = env::var("PATH_TO_WASM_BINARIES") {
+        PathBuf::from(custom_path)
+    } else {
+        let project_root = env::var("CARGO_MANIFEST_DIR").unwrap_or_else(|_| ".".to_string());
+        if cfg!(debug_assertions) {
+            PathBuf::from(project_root)
+                .join("../kairos-contracts/target/wasm32-unknown-unknown/debug/")
+        } else {
+            PathBuf::from(project_root)
+                .join("../kairos-contracts/target/wasm32-unknown-unknown/release/")
+        }
+    };
+    if !base_path.exists() {
+        let build_type = if cfg!(debug_assertions) {
+            "debug"
+        } else {
+            "release"
+        };
+        panic!("WASM directory does not exist: {}. Please build smart contracts at `./kairos-contracts` with `cargo build` for {}.", base_path.display(), build_type);
```

Comment:
This is not very nice imo, but if it allows you to run the tests without needing to be in a nix devShell it's fine to me.